sdk/servicebus/azure-servicebus/azure/servicebus/management/_management_client.py

```diff
@@ -683,6 +684,7 @@ def get_rule(self, topic, subscription, rule_name, **kwargs):
                 "Rule('Topic: {}, Subscription: {}, Rule {}') does not exist".format(
                     subscription_name, topic_name, rule_name))
         rule_description = RuleDescription._from_internal_entity(rule_name, entry.content.rule_description)
+        deserialize_rule_key_values(entry_ele, rule_description)
```

Comment:
in the PR I observe the pattern that all the `deserialize_rule_key_values` are called after `RuleDescription._from_internal_entity`, would we consider would the call into `RuleDescription._from_internal_entity`?

Reply:
This method needs the xml Element object. `RuleDescription._deserialize_rule_from_internal_entity` doesn't accept an xml Element object, like other XXXDescription._deserialize_yyy().
Extracting the extra information from the xml Element is workaround code. I would like to make them as decoupled from the main code as possible.

Reply:
gotcha, then I'm good with it for the sake of consistency.

and one more question for my own understanding:
in the future  if autorest get it supported then `deserialize_rule_key_values` is no longer needed?

Reply:
Right, deserialize_rule_key_values wouldn't be needed.